jobs/async-upload/job/upload.py

```diff
@@ -4,58 +4,60 @@
 from model_registry import utils
 from model_registry.utils import OCIParams, S3Params, save_to_oci_registry
 
+from .models import AsyncUploadConfig, DestinationConfig, OCIStorageConfig, S3StorageConfig
+
 logger = logging.getLogger(__name__)
 
-def _get_upload_params(config: Dict[str, Any]) -> S3Params | OCIParams:
+def _get_upload_params(config: AsyncUploadConfig) -> S3Params | OCIParams:
     """
     Returns the upload params for the destination type
 
     Args:
         config: Configuration dictionary
     """
-    logger.debug("🔍 Getting upload params for destination type: %s", config["destination"]["type"])
-    destination_config = config["destination"]
-    if destination_config["type"] == "s3":
+    destination_config = config.destination
+    logger.debug("🔍 Getting upload params for destination type: %s", destination_config)
+    if isinstance(config.destination, S3StorageConfig):
         return S3Params(
-            bucket_name=destination_config["s3"]["bucket"],
-            s3_prefix=destination_config["s3"]["key"],
-            endpoint_url=destination_config["s3"]["endpoint_url"],
-            access_key_id=destination_config["s3"]["access_key_id"],
-            secret_access_key=destination_config["s3"]["secret_access_key"],
-            region=destination_config["s3"]["region"],
+            bucket_name=config.destination.bucket,
+            s3_prefix=config.destination.key,
+            endpoint_url=config.destination.endpoint_url,
+            access_key_id=config.destination.access_key_id,
+            secret_access_key=config.destination.secret_access_key,
+            region=config.destination.region,
         )
```

Comment:
This is representative of most of the changes happening in the repo. Changing a dict-key-lookup for a proper dot-notation lookup